hazelcast/src/main/java/com/hazelcast/client/config/ClientRoutingMode.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.config;
+
+import com.hazelcast.spi.annotation.Beta;
+
+/**
+ * Represents the routing modes of the client for its invocations.
+ * @since 5.3
+ */
+@Beta
+public enum ClientRoutingMode {
+    /**
+     * The client will route all invocations to a single member,
+     * using that member as a proxy.
+     * <p>
+     * For partition and target specific invocations, if that member
+     * is not the owner of the partition or is not the target, the
+     * invocation will be routed the to correct member in the
+     * server-side. That adds two extra hops before the invocation
+     * returns a response to the user.
+     * <p>
+     * Using the uni-socket mode, the client will only have one
+     * active connection all the time. It might be useful in the
+     * presence of firewalls, security configurations, or some custom
+     * networking issues. Also, it might make sense if there are too
+     * many clients and one wants to avoid those clients to connect
+     * to all the members.
+     */
+    UNISOCKET((byte) 0),
```

Comment:
If we are going to introduce a new enum, I'm going to suggest using less crappy names than unisocket and multisocket. These names are really terrible.

Reply:
More appropriate names:

1) random member
2) right member
3) right core

Something along these lines.

Reply:
That makes sense 👍 Let me discuss that with the team as well